test/components/selection-list.spec.tsx

```diff
@@ -0,0 +1,128 @@
+
+        await waitForDom(() => {
+            expect(select('FOOD', 'LIST')).to.be.present();
+        });
+
+        simulate.click(select('FOOD', 'LIST')!.children[4]);
+
+        await waitForDom(() => {
+            expect(select('FOOD', 'RESULT')).to.contain.text('Sausage');
+        });
+    });
+
+    it("Works with a custom renderer and data schema", async () => {
+        const {select, waitForDom} = clientRenderer.render(<SelectionListDemo />);
+
+        await waitForDom(() => {
+            expect(select('EMOJI', 'LIST')).to.be.present();
+        });
+
+        const octopus = select('EMOJI', 'LIST')!.children[3] as HTMLElement;
+
+        expect(octopus).to.be.present();
+        expect(octopus.dataset.value).to.be.equal('Octopus');
+        expect(octopus).to.contain.text('🐙');
+        simulate.click(octopus);
+
+        return waitForDom(() => {
+            expect(select('EMOJI', 'RESULT')).to.contain.text('octopus');
+        });
+    });
+
+    it("Renders items under each other using the default renderer", async () => {
+        const {select, waitForDom} = clientRenderer.render(
+            <SelectionList dataSource={['One', 'Two', divider, 'Three']} />
+        );
+
+        return waitForDom(() => {
+            const items = Array.from(select('LIST')!.children);
+            expect(items).to.be.inVerticalSequence();
+            expect(items).to.be.horizontallyAligned('left');
+        });
+    });
+
+    it("Fires onChange when an item is clicked", async () => {
+        const dataSource = ['One', 'Two', 'Three'];
+        const onChange = sinon.spy();
+        const {select, waitForDom} = clientRenderer.render(
+            <SelectionList dataSource={dataSource} value="One" onChange={onChange} />
+        );
+
+        await waitForDom(() => {
+            expect(select('LIST')).to.be.present();
+        });
+
+        simulate.click(select('LIST')!.children[1]);
+
+        return waitFor(() => {
+            expect(onChange).to.have.been.calledOnce;
+            expect(onChange).to.have.been.calledWithExactly('Two');
+        }, 100);
+    });
+
+    it("Doesn't fire onChange for clicks on active items, dividers, disabled items, and items without value", async () => {
+        const dataSource = [
+            'One',
+            divider,
+            {label: 'Two'},
+            {value: 'Three', label: 'Three', disabled: true}
+        ];
+        const onChange = sinon.spy();
+        const {select, waitForDom} = clientRenderer.render(
+            <SelectionList dataSource={dataSource} value="One" onChange={onChange} />
+        );
+
+        await waitForDom(() => {
+            expect(select('LIST')).to.be.present();
+        });
+
+        simulate.click(select('LIST')!.children[0]);
+        simulate.click(select('LIST')!.children[1]);
+        simulate.click(select('LIST')!.children[2]);
+        simulate.click(select('LIST')!.children[3]);
+
+        return waitFor(() => {
```

Comment:
It is problematic to test with waitFor() that something **didn't** happen. If the "thing" **will** happen with a little delay, the assertion will still pass immediately. Better way is to check `to.have.not.been.called` after some delay.